File: TaskManagement-Spring/src/main/java/com/ye/task/service/admin/AdminService.java
package com.ye.task.service.admin;

import com.ye.task.dto.CommentDto;
import com.ye.task.dto.TaskDto;
import com.ye.task.dto.UserDto;

import java.util.List;

public interface AdminService {

    List<UserDto> getUsers();

    TaskDto createTask(TaskDto taskDto);

    List<TaskDto> getAllTasks();

    void deleteTask(Long id);

    TaskDto getTaskById(Long id );

    TaskDto updateTask(Long id , TaskDto taskDto);

    List<TaskDto> searchTaskByTitle(String title);

    CommentDto createComment(Long taskId , String content );



    List<CommentDto> getCommentsByTaskId(Long taskId);
}

File: taskmanagementAngular/src/app/modules/admin/services/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';

const BASE_URL = 'http://localhost:8080/'

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private http: HttpClient) { }

  getUsers() : Observable<any>{
    return this.http.get( BASE_URL + "api/admin/users" ,
      {
        headers: this.createAuthorizationHeader()
      }
    )
  }

  createTask(taskDto : any ) : Observable<any> {
    return this.http.post( BASE_URL + "api/admin/create-task" , taskDto , {
      headers : this.createAuthorizationHeader()
    })
  }

  getAllTask(): Observable<any>{
    return this.http.get( BASE_URL + "api/admin/tasks"  , {
      headers: this.createAuthorizationHeader()
    });
  }

  deleteTask( id : number ) : Observable<any> {
    return this.http.delete( BASE_URL + "api/admin/task/" + id , {
      headers : this.createAuthorizationHeader()
    });
  }

  getTaskById( id : number ) : Observable<any> {
    return this.http.get( BASE_URL + "api/admin/task/" + id  , {
      headers : this.createAuthorizationHeader()
    })
  }

  updateTask( id : number , taskDto : any ) : Observable<any> {
    return this.http.put( BASE_URL + `api/admin/task/${id}` , taskDto , {
      headers : this.createAuthorizationHeader()
    })
  }

  private createAuthorizationHeader():HttpHeaders{
    return new HttpHeaders().set(
      'Authorization' , 'Bearer ' + StorageService.getToken()
    )
  }
}
